### DISCUSSION TITLE
MediaFusion Docker Deployment Guide Not available on cloud server

### DISCUSSION THREAD

**Discussion by ymcoming** [mhdzumair/MediaFusion#172] [General]
I followed the MediaFusion Docker Deployment Guide and deployed it on oracle vps several times but failed. It seems that this guide only considers local deployment and does not consider remote VPS deployment. Can you update the guide? Thanks!

**Comment by mhdzumair** [comment]
Sure only thing that you need to purchase a Domain name for cheap to setup it correctly. will add the documentation in future.

**Reply by ymcoming** [reply]
I have a domain name and now I'm not sure what information to modify to make it work smoothly.
I modified these lines in /deployment/docker-compose/nginx.conf:
     server {
         listen 443 ssl;
         server_name mediafusion.local;     --->**mysite.com**
         ssl_certificate /etc/ssl/certs/mediafusion.local.pem; --->**mysite.com.pem**
         ssl_certificate_key /etc/ssl/private/mediafusion.local-key.pem; --->**mysite.com.key.pem**
         ssl_session_cache shared:SSL:10m;
         ssl_session_timeout 10m;

         location/{
             proxy_pass http://mediafusion:8000; --->**127.0.0.1:8000**


Also modified the first line of deployment/docker-compose/.env:
HOST_URL="https://mediafusion.local"     --->**mysite.com**


But https://mysite.com still cannot be opened when running, and each docker is in the start state.

**Comment by srejithn** [comment]
I'm looking for this documentation too. It will be very helpful if you add this documentation. 
great work 
thanks

**Comment by mhdzumair** [comment]
To set up the server with DNS name + Cloudflare + docker compose:

While following the docker compose file update the nginx.conf with your server Domain name. 
- replace the key file name "mediafusion.local" into your domain name. 
- Do the same thing in docker-compose.yml file 
- then generate self signed key with mkcert 
- after running the mediafusion containers, make sure to whitelist the port 443 in the server security group. in some server like azure, you need to whitelist the port 443 through ip-table rule. 
- Now goto your cloudflare dashboard and set SSL mode Full as shown. 
![image](https://github.com/mhdzumair/MediaFusion/assets/44891939/da5bc8cf-adf8-49f7-a0b0-4ad637c37736)
- Update the DNS for @ to point out to your server IP address. 
![image](https://github.com/mhdzumair/MediaFusion/assets/44891939/9410741f-9f92-4edc-b6f4-9e0c1ac61f26)

Thats it. Now see if this is work. If you face any issue let me know.

**Reply by srejithn** [reply]
It worked. Running perfectly. Great work.thanks buddy.

**Comment by ymcoming** [comment]
My deployment was successful, thank you!
But an error occurred:
Error response from daemon: manifest for mhdzumair/mediafusion:v3.8.8 not found

After I changed it to 3.8.1, it ran normally.

**Reply by ymcoming** [reply]
"The Docker method is not very stable, and I want to learn to deploy Minikube. What changes are needed to deploy Minikube on a VPS? Thank you!"

**Comment by hackadelic** [comment]
best method to add own domain if not using cloudflare? like on a dedi or vm with full connectivity no nat or anything.. is pointing my own reverse proxy to mediafusion:8000 okay? i can access the webui and everything but all of my streams are showing playback error: please try again so im not sure whats going on, server is open from the internet and i have a caddy reverse proxy pointed to localhost:8000 instead of the nginx and its showing the webui perfectly, i even set up the addon and add it into stremio without issue, but playback fails for every debrid provider, any and all help or suggestions would be super duper helpful (:
